File: client/src/game/conf/ItemConfig.ts
class ItemConfig {
    private static _dic:Dictionary<ItemConfig> = new Dictionary<ItemConfig>();
    private static _arr:Array<ItemConfig> = new Array<ItemConfig>();

	 public Id :number ;
     public Name :string ;
     public ItemType :number ;
     public Icon :string ;
     public Star :number ;
     public AddType :number ;
     public WorthPrice :number ;
     public Iconframe :string ;
     public IconScale :number ;
    

    
    public constructor(){

    }


    public static get Dic():Dictionary<ItemConfig>{
        return  ItemConfig._dic;
    }
    public static get Arr():Array<ItemConfig>{
        return ItemConfig._arr;
    }
    public static parse(json:any[]){
        if(json && json.length > 0){
			for(var i = 0; i < json.length; i++){
				let config:ItemConfig = <ItemConfig>json[i];
				ItemConfig._arr.push(config);
				ItemConfig._dic.add(config.Id,config);
			}
        }else{
            console.error("ItemConfig loader fail!");
        }
    }
}

File: server/src/game/conf/ItemConfig.go
package conf

import (
	//"bufio"
	"encoding/json"
	"fmt"
	//"io"
	"io/ioutil"
	"os"
	//"strings"
)

type ItemConfigElem struct {
	Id int
	Name string
	ItemType int
	Icon string
	Star int
	AddType int
	WorthPrice int
	Iconframe string
	IconScale float32
	
}

type ItemConfig0 struct {
	Dic map[int]*ItemConfigElem
	Arr []*ItemConfigElem
}

var (
	ItemConfig ItemConfig0
)

func loadItemConfig(path string) {
	file, err := os.Open(path)
	defer func() {
		if file != nil {
			file.Close()
		}
	}()
	if err != nil {
		panic(fmt.Sprint("load config error, path=", path))
	}

	ItemConfig.Dic = make(map[int]*ItemConfigElem)
	ItemConfig.Arr = make([]*ItemConfigElem, 0, 32)

	bt,err  :=  ioutil.ReadAll(file)
	json.Unmarshal(bt, &ItemConfig.Arr)
	for _,conf := range ItemConfig.Arr{
		ItemConfig.Dic[conf.Id] = conf
	}
	
}
